### DISCUSSION TITLE
Plausibilität von Check-Ins automatisiert prüfen

### DISCUSSION THREAD

**Discussion by drive-n-code** [Traewelling/traewelling#2764] [Ideas / Feature Requests]
Es kommt wohl vor, dass sich Leute auf Züge einchecken, nur um eingecheckt zu sein, aber gar nicht (vor haben) mit(zu)fahren.

Um "Punkte-Sammel-Missbrauch" o.Ä. zu vermeiden, wäre meine Idee Check-Ins auf Plausibilität zu prüfen.

- Neue Einstellung in Traewelling: "Punkte für Check-Ins sammeln (JA/NEIN)"
- Wenn Punkte sammeln aktiviert ist, muss eine Standort-Berechtigung für Traewelling im Browser aktiviert sein
- Wenn Punkte sammeln aktiviert ist, wird beim Check-In geprüft, ob die Position des Nutzers für den Check-In plausibelt ist (Je nach Zeit bis zur/seit der Abfahrtszeit verglichen mit der Position des Einstieghaltes)
- Wenn Punkte sammeln aktiviert ist und der Check-In nicht plausibel sein sollte, den Nutzer darauf hinweisen und fragen ob trotzdem eingecheckt werden soll (bei dem Check-In werden dann keine Punkte gutgeschrieben).

**Comment by marhei** [comment]
Spätestens mit Drittapps über die API lässt sich nicht mehr überprüfen wie legit die gesendete GPS-Location ist und gleichzeitig widerspricht das jeglicher Datensparsamkeit.

**Comment by HerrLevin** [comment]
Die Einstellung des Punkte-Sammelns gefällt mir und ich behalte das mal im Hinterkopf. 

Ansonsten bin ich leider bei Marcel. Es ist auch einfach nicht im Sinne eines freundlichen user-flows.

Alternativ könnte man einfach globale Statistiken deaktivieren. Siehe #2759

**Comment by drive-n-code** [comment]
Ich finde nicht unbedingt, dass das der Datensparsamkeit widerspricht.
Die Locations werden nicht gespeichert. Und wer seine echte Location nicht preisgeben möchte, kann es recht einfach dauerhaft deaktivieren.

Der User-Flow wird meiner Meinung nach dadurch auch nicht beeinträchtigt.
Es ist eine einmalig Sache entweder das Punkte sammeln abzuschalten oder eben die Berechtigung zu geben, dass die Location abgefragt werden darf (in dem Fall pro Endgerät).

Es wäre nur die Frage ob man das Ganze Serverseitig (und stringent prüft) oder ausschließlich Clientseitig prüft (und man damit die API nicht aufbläht) - ganz nach dem Motto "Wer es drauf anlegt und technisch versiert ist, findet eh immer einen Weg sich zu "pushen".


Als Alternative die globalen Statistiken zu deaktivieren fänd ich eigentlich nicht so toll. Es ist immer ganz nett da mal reinzuschauen. Oder sich daran zu erfreuen, wenn man mal viel unterwegs ist und dort auch mal schafft aufzutauchen. Hier könnte man eher gucken, ob man die Statistiken etwas "aufbohrt". Bspw. daraus eine Top 50 macht oder einen Reiter hinzufügt á la "Privat unterwegs". Aber das ist eine andere Diskussion.

**Reply by HerrLevin** [reply]
Die Freigabe der Location funktioniert auf iOS im Browser leider nicht dauerhaft. Das funktioniert also nicht richtig, außer ich frage den user bei jedem Seitenaufruf neu.

Außerdem erlauben wir derzeit "straffreie" Checkins 60 Minuten vor/nach Abfahrt/Ankunft. Das würde es also auch ziemlich einschränken.

**Reply by drive-n-code** [reply]
Dass das unter iOS nicht dauerhaft geht wusste ich nicht. Das ist dann natürlich schon arg einschränken. :/

Die straffreien Check-Ins könnte man aber beibehalten. Im Rahmen eines zeitabhängigen in der größe variierenden Georadius von Start/Zielhaltestelle, bzw. Position vom Streckenabschnitt, auf dem Zug gerade unterwegs ist (falls man erst während der Fahrt eincheckt).
Ja, ich weiß, ganz trivial wäre diese Logik natürlich nicht. Aber an sich machbar.

**Comment by pascalwittler** [comment]
**Nichtverfügbarkeit von GPS**

Auf Smartphones und den dort weit verbreiteten Webbrowsern mag eine GPS-Positionierung mit fast allen Produkten und hinreichender Genauigkeit verfügbar sein, mit diversen Einschränkungen und Mängeln in der Nutzerfreundlichkeit wie oben erwähnt z.B. unter iOS/Safari.

Da die Nutzung von Träwelling aber auf allen Geräten mit einem einigermaßen glaubwürdigen Webbrowser möglich ist, z.B. auch (mein persönlich häufiger Anwendungsfall) auf Notebooks, die i.d.R. keine Hardware-Ortungsfunktion haben und/oder die Positionierung, die der Webbrowser ersatzweise annimmt, völlig unpräzise z.B. am nächsten Mobilfunkmasten oder einem willkürlichen kabelgebundenen Standort des ISP erfolgt, stellt sich für mich die Frage, ob durch eine Einführung des hier geforderten Features die freie Geräte- und Browserwahl nicht in Frage gestellt wird.

Beispielsweise schreibe ich diesen Kommentar gerade mit meinem Notebook in der Karlsruher Südweststadt, wenn ich aber meinen Webbrowser (bzw. Chrome, Grund s.u.) frage, wo ich seiner Meinung nach bin, wird mir ein Ort nahe Rastatt angezeigt.

Weiterhin gibt es Fälle, in denen Geräte und Webbrowser die letzte gemessene GPS-Position weiterverwenden, wenn die GPS-Funktion gerade nicht verfügbar ist. Ebenso ist es möglich (wie ich das in meinem Firefox gemacht habe), die Ortung als Feature generell auszuschalten.

Unabhängig davon kann ich mindestens in Firefox in den Entwicklertools eigene Koordinaten manuell eingeben, um z.B. von mir entwickelte Webanwendungen zu testen. Ähnliche Features sind mir unter Android mit aktiviertem Entwicklermodus bekannt.

Es ergeben sich also unterschiedlichste Szenarien, in denen Menschen keine Ortung als "Beweis" vornehmen können oder wollen. Menschen das allein als Umstand während eines Checkins auf dem Bildschirm zu vermitteln, dürfte die Anwendung deutlich nutzerunfreundlicher machen. Menschen, die aus welchem Grund auch immer so scharf auf Punkte sind, dass sie "Fake"-Checkins vornehmen, haben verschiedene Möglichkeiten, sich durchzumogeln.

**Comment by pascalwittler** [comment]
**Genauigkeit**

Besonders mit Blick auf ÖPNV innerhalb von Großstädten mit vielen Haltestellen auf engem Raum stellt sich die Frage, wie präzise die GPS-Positionierung eingestellt werden soll.

Allein in Karlsruhe sind Bahnsteiglängen von 160 Metern und größere Abstände zwischen Bahnsteigen der selben Station in verschiedene Richtungen nicht mehr unüblich. In Träwelling zusammengefasste Stationen wie Karlsruhe Hbf (Vorplatz Bahn), Karlsruhe Hbf (Vorplatz Bus / Messe-Express / …), Karlsruhe Hbf (DB-Gleise 1-14 + 101/102), Karlsruhe Hbf (Südausgang), Karlsruhe Hbf (SEV) haben schnell eine Gesamtfläche von mehreren hundert Quadratmetern. Der Abstand zwischen Stationen besteht aber selbst im Umland dank Karlsruher Modell oft aus nur 300 bis 400 Metern.

Zusammen mit Puffer für Ungenauigkeiten bei der Positionsbestimmung (speziell in Gebäuden, unterirdisch usw.), Verzögerungen beim Aktualisieren der GPS-Daten usw. usw. ergibt sich hier für mich ein Widerspruch bei der hier angedachten Plausibilitätsprüfung, die entweder diverse "ehrliche" Checkin fälschlicherweise "bestraft" – oder Leute mit starken Ambitionen zu "Fake"-Punkten buchen einfach im Rahmen der notwendigerweise eingeräumten Toleranz ihres Standortes 20 Straßenbahn- oder Busfahrten von 500 Metern hin und her.

**Comment by pascalwittler** [comment]
**Umleitungen, Störungen**

Es soll in der Erfolgsgeschichte der Deutschen Eisenbahn schon Tage gegeben haben, an denen Züge kurzfristig umgeleitet wurden oder Start- und Zielbahnhof sich z.B. innerhalb einer Stadt oder auch in einer ganzen Region geändert haben. Es stellt sich also die Frage, was passiert, wenn Zugdaten in Träwelling (noch) nicht 100% mit den Echtzeitdaten des tatsächlichen Betriebs übereinstimmen und ich z.B. in Bonn-Beuel statt Bonn Hbf, in Ingolstadt statt Augsburg oder in Karlsruhe-Durlach statt Karlsruhe Hbf einchecken muss, um nur einige Beispiele aus den letzten Wochen zu nennen.

Werde ich dann "bestraft", weil meine GPS-Position nicht mit den (Soll-)Fahrplandaten übereinstimmt und/oder weil manuell (z.B. live vom Zugpersonal) gepflegte ausgefallene und Ersatzhalte nicht perfekt gematcht werden? Oder soll andersherum die GPS-Position so grob und weitläufig ausgelegt werden, dass z.B. die o.g. Änderungen noch als plausibel gelten, dafür aber wieder so viele "Fake"-Perspektiven offen bleiben, dass man sich die Prüfung gleich sparen kann?

**Comment by pascalwittler** [comment]
**Checkins vor, während oder nach einer Fahrt**

Aktuell sind Checkins einige Minuten vor, die ganze Zeit während und einige Minuten nach der Fahrt möglich, ohne direkt alle Punkte abgezogen zu bekommen. Hierdurch ergeben sich diverse Szenarien mit Auswirkungen auf die GPS-Positionierung zum Checkin-Zeitpunkt:

* Checkin 20 Minuten vor Abfahrt aus meinem vorherigen Zug drei Stationen vor dem Beginn meiner eingecheckten Fahrt oder auf dem Fußweg zum Bahnhof
* Checkin während der Fahrt – also entweder unplausibel weil nicht mehr im GPS-Bereich des Startbahnhofs oder nur plausibel anhand der (wie in anderen Diskussionen und Issues deutlich) nicht immer präzisen Routenberechnung oder unplausibel bei kurzfristig geänderter Route des Zuges (z.B. Streckenstörung, Umleitung linke/rechte Rheinstrecke, Umleitung Vaihingen(Enz)–Stuttgart über Bietigheim-Bissingen, …)
* Checkin nach Ankunft – entweder unplausibel weil nicht mehr im GPS-Bereich des Startbahnhofs oder unplausibel, weil gesamte Strecke der Zugfahrt schon verlassen, mit anderem Zug einige km weitergefahren etc. etc.

**Comment by pascalwittler** [comment]
**Fake des Zielbahnhofs**

Im o.g. Featurewunsch ist von einer einmaligen GPS-Positionierung zum Checkin-Zeitpunkt die Rede, was insofern Sinn macht, als dass man die Position nur als Momentaufnahme heranziehen kann. Was aber, wenn eine ganz ganz böse Person heruausfindet, dass die Position am Startbahnhof stimmen muss, um Punkte für die Fahrt zu bekommen, deshalb auch in Leipzig Hbf plausibel eincheckt, aber als Zielbahnhof, um mehr Punkte zu bekommen, nicht Erfurt eingibt, sondern Emden Außenhafen?

**Reply by drive-n-code** [reply]
Gänzlich verhindern kann man das nie. Wer "betrügen" will, schafft es auch.
Es geht nur darum die Hürde für Otto-Normal Nutzer zu erhöhen, möglichst ohne Einbußen im normalen Gebrauch von Traewelling. Das war die Idee.

**Reply by pascalwittler** [reply]
Einfach im Dropdown-Menü eine Auswahlmöglichkeit weiter unten anklicken, halte ich für keine Hürde für Otto-Normal-Nutzer.
